examples/unzer-payment/composables/useUnzer.ts

```diff
@@ -0,0 +1,70 @@
+import type { Schemas } from "#shopware";
+import type { Ref } from "vue";
+
+declare let unzer: UnzerSdkConstructor;
+
+export interface UnzerSdk {}
+
+export interface UnzerSdkConstructor {
+  new (
+    publicKey: string,
+    unzerInstanceOptions: { locale: string } | null,
+  ): UnzerSdk;
+}
+
+/**
+ * Composable for Unzer payments
+ *
+ * @public
+ */
+export function useUnzer(): {
+  isUnzerPaymentMethod: (paymentMethod: Schemas["PaymentMethod"]) => boolean;
+  unzerInstance: Ref<UnzerSdk | null>;
+} {
+  const isUnzerPaymentMethod = (paymentMethod: Schemas["PaymentMethod"]) =>
+    typeof paymentMethod?.shortName === "string" &&
+    paymentMethod.shortName.includes("unzer");
+
+  const unzerInstance = ref<UnzerSdk | null>(null);
+
+  function createUnzerInstance() {
+    if (
+      unzerInstance.value === null ||
+      typeof unzerInstance.value === "undefined"
+    ) {
+      const runtimeConfig = useRuntimeConfig();
+      const publicKey = runtimeConfig.public.unzer?.publicKey;
+
+      if (!publicKey) {
+        throw new Error("[useUnzer] public key is not defined");
+      }
+
+      const { $i18n } = useNuxtApp();
+      unzerInstance.value = new unzer(publicKey, {
+        locale: $i18n.locale.value,
+      });
+    }
+  }
+
+  onBeforeMount(() => {
+    useHead({
```

Comment:
attaching the unzer.js script is too lazy. It's not being loaded in the test env: https://stackblitz.com/edit/github-c9t5on?file=nuxt.config.ts (backend is temporary, we can talk about it in DM later on)

it causes the error in here: https://github.com/shopware/frontends/pull/671/files#diff-a73f8d424b5c2834bf69de34a0cd276893eb9a66a9a84ef96553079f816ab0e1R49

I can see two solutions for that:
- I would wrap useHead things in a function and export it in the composable to be used directly in the setup function of the component - then run it in `onBeforeMount` lifecycle hook in a component.
- use a plugin (server + client) to register a script for specific page, in client plugin initialize the unzer instance

wdyt?